src/main.cpp

```diff
@@ -15,24 +15,34 @@
 #include <chrono>
 #include <fstream>
 #include <unistd.h>
+#include <exception>
 
 int main(int argc, char* argv[])
-{
+{   
     std::string input_name;
     int k;
-    int opt;
+    int opt, prev_ind;;
     int verbosity = 0;
     std::string loss = "2";
+    bool f_flag = false;
+    bool k_flag = false;
 
-    while ((opt = getopt(argc, argv, "f:l:k:v:")) != -1) {
+    while (prev_ind = optind, (opt = getopt(argc, argv, "f:l:k:v:")) != -1) {
```

Comment:
I introduced `prev_ind` to handle the case when we have flag without argument but followed by another flag which is then assumed as the argument of the previous flag. E.g. `./BanditPAM -f -k 5` will give an error of missing `k` flag which is not desired.

Refer to https://www.gnu.org/software/libc/manual/html_node/Using-Getopt.html, `optind` is a variable set by `getopt` to the index of the next element of the argv array to be processed. Initially, `optind` is set to `1` and thus the same as `prev_ind`. 

When we loop through the flag, if the option was the last character in the string pointed to by an element of `argv`, then `optarg` shall contain the next element of `argv`, and `optind` shall be incremented by `2`. So in the previous example, the result will be `optind = 3`, `prev_ind = 1` and `*optarg = '-'`. This condition means there is something like `-f -k` in the input and thus we assign `opt = ':'` which is a pre-defined case prompting `option needs a value`.

Since the `prev_ind` needs to read the previous position of `opt_ind` so I have to put it in the loop, if it is outside the loop the two variables will always be the same. In general, the `prev_ind` condition will only be triggered when something like `-f -k` happens.

Reply:
Ah ok, thank you for the reference and explanation. From the SO link below, it looks like this is an acceptable way to do this. Perhaps long-term we should move to boost as in the second recommended solution.